std/experimental/allocator/building_blocks/region.d

```diff
@@ -53,26 +53,34 @@ struct Region(ParentAllocator = NullAllocator,
     {
         alias parent = ParentAllocator.instance;
     }
+
+    static if (!is(ParentAllocator == NullAllocator))
+    {
+        void[] _innerStorage;
+    }
     private void* _current, _begin, _end;
 
+
     /**
-    Constructs a region backed by a user-provided store. Assumes $(D store) is
-    aligned at $(D minAlign). Also assumes the memory was allocated with $(D
-    ParentAllocator) (if different from $(D NullAllocator)).
+    Constructs a region backed by a user-provided store.
+    Assumes the memory was allocated with $(D ParentAllocator)
+    (if different from $(D NullAllocator)).
 
     Params:
-    store = User-provided store backing up the region. $(D store) must be
-    aligned at $(D minAlign) (enforced with $(D assert)). If $(D
-    ParentAllocator) is different from $(D NullAllocator), memory is assumed to
-    have been allocated with $(D ParentAllocator).
-    n = Bytes to allocate using $(D ParentAllocator). This constructor is only
-    defined If $(D ParentAllocator) is different from $(D NullAllocator). If
-    $(D parent.allocate(n)) returns $(D null), the region will be initialized
-    as empty (correctly initialized but unable to allocate).
-    */
+        store = User-provided store backing up the region. If $(D
+        ParentAllocator) is different from $(D NullAllocator), memory is assumed to
+        have been allocated with $(D ParentAllocator).
+        n = Bytes to allocate using $(D ParentAllocator). This constructor is only
+        defined If $(D ParentAllocator) is different from $(D NullAllocator). If
+        $(D parent.allocate(n)) returns $(D null), the region will be initialized
+        as empty (correctly initialized but unable to allocate).
+        */
     this(ubyte[] store)
     {
-        store = cast(ubyte[])(store.roundUpToAlignment(alignment));
+        static if (!is(ParentAllocator == NullAllocator))
```

Comment:
@andralex 
If provided with storage from a ParentAllocator, the Region would incorrectly call 'deallocate' from its destructor.

This constructor here changes the "store" range parameter, which is later passed to the ParentAllocator for deallocation, in the destructor.

Reply:
See below, I also added a test to reflect this change

Reply:
Good catch